test/GetOrderUseCase.test.ts

```diff
@@ -0,0 +1,31 @@
+import { GetOrderUseCase } from "../src/Application/GetOrderUseCase";
+import { Order } from "../src/Domain/Entity/Order";
+import { Product } from "../src/Domain/Entity/Product";
+import { Dimensions } from "../src/Domain/ValueObjects/Dimensions";
+import { Email } from "../src/Domain/ValueObjects/Email";
+import { Id } from "../src/Domain/ValueObjects/Id";
+import { OrderRepositoryMemory } from "../src/Infra/Repository/OrderRepositoryMemory";
+
+describe("GetOrderUseCase tests", () => {
+  it("should get order by id", async () => {
+    const orderRepositoryMemory = new OrderRepositoryMemory();
+    const order = new Order(new Id("1"), new Email("cliente@email.com"), new Date("2023-01-01T00:00:00"), 1);
+    const product = new Product(new Id("1"), "Fone de ouvido", 10.0, new Dimensions(10, 20, 30), 0);
+    order.addItem(product, 1);
+    await orderRepositoryMemory.save(order);
+
+    const getOrderUseCase = new GetOrderUseCase(orderRepositoryMemory);
+    const savedOrder = await getOrderUseCase.execute({id: "1"});
+
+    expect(order.id).toEqual(savedOrder.id)
+  });
+
+  it("should explode expcetion with invalid order id", async () => {
+    expect(async () => {
+      const orderRepositoryMemory = new OrderRepositoryMemory();
+
+      const getOrderUseCase = new GetOrderUseCase(orderRepositoryMemory);
+      await getOrderUseCase.execute({id: "Invalid ID"});
+    }).rejects.toThrow(new Error("Order not found"));
+  });
+});
```

Comment:
Refatorar teste:
* Corrigir descrição do teste para `should throw exception if order not found`
* Construir repositório + SUT de forma global (`beforeAll` ou `beforeEach`)
* Adicionar fazer a limpeza dos dados antes de cada teste (`beforeEach`)

Exemplo:
```typescript

describe("GetOrderUseCase tests", () => {
  const orderRepository = new OrderRepositoryMemory();
  const getOrderUseCase = new GetOrderUseCase(orderRepository);

  beforeEach(async () => {
    await orderRepository.clear():
  });

  it("should get order by id", async () => {
    const order = new Order(new Id("1"), new Email("cliente@email.com"), new Date("2023-01-01T00:00:00"), 1);
    const product = new Product(new Id("1"), "Fone de ouvido", 10.0, new Dimensions(10, 20, 30), 0);
    order.addItem(product, 1);
    await orderRepository.save(order);

    const savedOrder = await getOrderUseCase.execute({id: "1"});

    expect(order.id).toEqual(savedOrder.id)
  });

  it("should explode expcetion with invalid order id", async () => {
    expect(async () => {
      await getOrderUseCase.execute({id: "Invalid ID"});
    }).rejects.toThrow(new Error("Order not found"));
  });
});
```